### PR TITLE
fix: getMediaQuery sort was returning NaN, which behaved differently …

### LINKED ISSUE #1137: [General] Cross-browser inconsistencies in media-queries
Opening on behalf of pablo on Discord.

Breakpoints are differently computed, apparently based on browser?

**Expected Behavior** Breakpoints are consistent across browsers.

https://codesandbox.io/s/chakra-ui-media-issue-l1kex?file=/src/index.js

**Steps to Reproduce** 

Open in both Chrome and Firefox.
Chrome works as expected. XL > LG > MD > SM.
Firefox renders:
- MD above 1280px
- SM and LG below 1280px
- SM and XL below 992px
- SM below 768px

### PR DESCRIPTION
**Pull request by jmiazga**
Fixes #1137 

Ok so this was a fun one. The compare function in `sort` was returning `NaN`. That is because `'30em' - '48em'` results in `NaN`.

In Chrome nothing happened to the order of the array after `sort` was executed, but in Firefox the array was reversed.

### CODE REVIEW

File: packages/parser/src/utils/get-media-query.ts [hunk 1]
```diff
@@ -21,7 +21,12 @@ export function getMediaQuery(breakpoints?: Dict, mapper = toMediaQuery) {
 
   const asArray = objectKeys(_breakpoints)
     .map((key) => _breakpoints[key])
-    .sort((a, b) => a - b)
+    .sort((a, b) => {
+      const regex = /^\d+/
```
Comment (ljosberinn) [inline packages/parser/src/utils/get-media-query.ts:25]: I'm unaware under which circumstances this is called so please forgive me if this is wrong, but but I'd assume this is performance sensitive so we might be able to gain a small win here by having the regex outside of the compare function, or even extract the whole sort-function to its own definition?

Comment (with-heart) [inline packages/parser/src/utils/get-media-query.ts:25]: Agree with @ljosberinn, these changes are perfect if you at least define `regex` outside of the function call so it's only created once. I'm not entirely sure of the performance implications, but we might as well avoid it anyway!

**Comment by with-heart** [comment]
Interestingly enough, I know that @segunadebayo is reworking the `css` and `parser` packages into a `styled-system` package that reuses a lot of `styled-system` internals for performance reasons, and this functionality no longer exists in this form there.

You can see the new `css` function (which is where `getMediaQuery` was used previously) in the `benchmarks` branch: https://github.com/chakra-ui/chakra-ui/blob/benchmarks/packages/styled-system/src/css.ts

**Comment by jmiazga** [comment]
@with-heart Should I make the changes or will this PR bc closed bc of the work on the `benchmarks` branch?

**Comment by jmiazga** [comment]
Ok I pushed a fix that I like better than the original. Not sure the timing of the `benchmarks` branch, but this will at least fix array responsive prop issues that people are having.

**Comment by segunadebayo** [comment]
As mentioned in #1137. I'll close this in favor of using styled-system's core functions (not styled system itself)

**Pull request closed without merging**